src/android/unregisterNativeModule.js

```diff
@@ -4,30 +4,32 @@ const path = require('path');
 const compose = require('lodash').flowRight;
 const getReactVersion = require('../getReactNativeVersion');
 const getPrefix = require('./getPrefix');
+const makeSettingsPatch = require('./patches/makeSettingsPatch');
+const makeBuildPatch = require('./patches/makeBuildPatch');
+
+const isInstalled = require('./isInstalled');
 
 const cut = (scope, pattern) =>
   scope.replace(pattern, '');
 
 module.exports = function unregisterNativeAndroidModule(name, dependencyConfig, projectConfig) {
   const prefix = getPrefix(getReactVersion(projectConfig.folder));
 
+  const settingsPatch = makeSettingsPatch(name, dependencyConfig, {}, projectConfig).patch;
+  const buildPatch = makeBuildPatch(name).patch;
+
   /**
    * @param  {String} content Content of the Settings.gradle file
    * @return {String}         Patched content of Settings.gradle
    */
-  const cutModuleFromSettings = (name) => (content) =>
-    cut(content, `include ':${name}'\n` +
-      `project(':${name}').projectDir = ` +
-      `new File(rootProject.projectDir, '../node_modules/${name}/android')\n`
-    );
+  const cutModuleFromSettings = (content) => cut(content, settingsPatch);
 
   /**
    * Cut module compilation from the project build
    * @param  {String} content Content of the Build.gradle file
    * @return {String}         Patched content of Build.gradle
    */
-  const cutModuleFromBuild = (name) => (content) =>
-    cut(content, `    compile project(':${name}')\n`);
+  const cutModuleFromBuild = (content) => cut(content, buildPatch);
 
   const getAddPackagePatch = require(`./${prefix}/addPackagePatch`);
```

Comment:
That does not work, but didn't do it because of params (we have to do something like replace all params in a packageInstance to `s+` and construct a regexp expression `new Regexp` to use it instead)